buildSrc/src/main/java/com/google/firebase/gradle/plugins/PomValidator.kt

```diff
@@ -16,63 +16,77 @@ package com.google.firebase.gradle.plugins
 
 import java.io.File
 import java.net.URL
-import javax.xml.parsers.DocumentBuilder
 import javax.xml.parsers.DocumentBuilderFactory
 import org.gradle.api.DefaultTask
 import org.gradle.api.GradleException
 import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputFile
 import org.gradle.api.tasks.TaskAction
-import org.w3c.dom.Document
 import org.w3c.dom.Element
-import org.w3c.dom.Node
-import org.w3c.dom.NodeList
 
+/**
+ * Ensures that pom dependencies are not accidently downgraded.
+ *
+ * Compares the latest pom at gmaven for the given artifact with the one generate for the current
+ * release.
+ *
+ * @throws GradleException if a dependency is found with a degraded version
+ *
+ * @property pomFile The pom file for the current release
+ * @property artifactId The artifactId for the pom parent
+ * @property groupId The groupId for the pom parent
+ */
 abstract class PomValidator : DefaultTask() {
-  @get:InputFile abstract val pomFilePath: Property<File>
+  @get:InputFile abstract val pomFile: Property<File>
```

Comment:
Would it be better to use https://docs.gradle.org/current/javadoc/org/gradle/api/file/RegularFileProperty.html?